dora/shell/src/main/java/alluxio/cli/CheckCluster.java

```diff
@@ -0,0 +1,236 @@
+  public static final String ANSI_RED = "\u001B[31m";
+  public static final String ANSI_GREEN = "\u001B[32m";
+  private static final List<BlockWorkerInfo> FAILED_WORKER = new ArrayList<>();
+  private static List<BlockWorkerInfo> sWorkerInfoList;
+  private static WorkerLocationPolicy sPolicy;
+
+  private CheckCluster() {
+  } // Prevent instantiation
+
+  /**
+   * Main function to test the cluster setup.
+   *
+   * @param args command line arguments (none expected)
+   */
+  public static void main(String[] args) {
+    setupConfiguration();
+
+    try (FileSystemContext fsContext = FileSystemContext.create(Configuration.global())) {
+      FileSystem fs = FileSystem.Factory.create(fsContext);
+      sWorkerInfoList = fsContext.getCachedWorkers();
+
+      // Test with caching
+      try {
+        runTestWithCaching(fs);
+      } catch (Exception e) {
+        System.out.println("Exception occurred during cache through test.");
+        handleException(e);
+      }
+
+      System.out.println("-----------------------");
+
+      // Test without caching (only UFS)
+      try {
+        runTestWithoutCaching(fs, fsContext);
+      } catch (Exception e) {
+        System.out.println("Exception occurred during through test.");
+        handleException(e);
+      }
+    } catch (Exception e) {
+      System.out.println(
+          "Exception occurred during FileSystemContext creation or worker retrieval.");
+      handleException(e);
+    }
+  }
+
+  private static void handleException(Exception e) {
+    e.printStackTrace();
+    System.out.println("Exception details: " + e.getMessage());
+  }
+
+  private static void setupConfiguration() {
+    Configuration.set(PropertyKey.USER_WORKER_SELECTION_POLICY,
+        "alluxio.client.file.dora.ConsistentHashPolicy");
+    Configuration.set(PropertyKey.DORA_CLIENT_UFS_FALLBACK_ENABLED, false);
+  }
+
+  private static AlluxioURI initializeTestDirectory(FileSystem fs) throws Exception {
+    FileSystem underlyingFileSystem = ((DoraCacheFileSystem) fs).getUnderlyingFileSystem();
+    String testDirName = Random.class.getSimpleName() + System.currentTimeMillis();
+    AlluxioURI testDir = new AlluxioURI(DIRECTORY + testDirName);
+    DoraCacheFileSystem doraFs = (DoraCacheFileSystem) fs;
+    testDir = doraFs.convertToUfsPath(testDir);
+    if (!underlyingFileSystem.exists(testDir)) {
+      underlyingFileSystem.createDirectory(testDir);
+    }
+    return testDir;
+  }
+
+  private static HashMap<BlockWorkerInfo, AlluxioURI> assignToWorkers(FileSystem fs,
+                                                                      AlluxioURI testDir,
+                                                                      boolean throughOnly)
+      throws Exception {
+    HashMap<BlockWorkerInfo, AlluxioURI> workerMap = new HashMap<>();
+    sPolicy = WorkerLocationPolicy.Factory.create(Configuration.global());
+
+    for (int i = 0; i < NUM_FILES && workerMap.size() < sWorkerInfoList.size(); i++) {
```

Comment:
1. what if not all workers are covered after NUM_FILES-th iterations? 
2. can you make this one configurable and recommend the value for different numbers of workers? (assume data are evenly distributed across the hash ring)